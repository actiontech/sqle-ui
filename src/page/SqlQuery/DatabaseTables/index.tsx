import DatabaseTables from './DatabaseTables';

export type DatabaseTablesProps = {
  dataSourceName?: string;
  schemaName?: string;

  getTableSchema: (tableName: string) => Promise<void>;
};

export default DatabaseTables;
