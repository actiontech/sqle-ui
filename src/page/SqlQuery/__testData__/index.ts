import {
  IGetSQLResultResDataV1,
  IInstanceResV1,
  IPrepareSQLQueryResDataV1,
} from '../../../api/common';
import { SqlQueryResultType } from '../index.type';

export const instance_info: IInstanceResV1 = {
  sql_query_config: {
    max_pre_query_rows: 50,
  },
};

export const query_ids_test1: IPrepareSQLQueryResDataV1 = {
  query_ids: [
    {
      query_id: 'query_id1',
      sql: 'select * from user1',
    },
    {
      query_id: 'query_id2',
      sql: 'select * from user2',
    },
    {
      query_id: 'query_id3',
      sql: 'select * from user3',
    },
    {
      query_id: 'query_id4',
      sql: 'select * from user4',
    },
  ],
};

export const query_res_1: IGetSQLResultResDataV1 = {
  current_page: 1,
  end_line: 16,
  execution_time: 123,
  head: [
    {
      field_name: 'field1',
    },
    {
      field_name: 'field2',
    },
    {
      field_name: 'field3',
    },
    {
      field_name: 'field4',
    },
    {
      field_name: 'field5',
    },
    {
      field_name: 'field6',
    },
    {
      field_name: 'field7',
    },
    {
      field_name: 'field8',
    },
    {
      field_name: 'field9',
    },
    {
      field_name: 'field10',
    },
    {
      field_name: 'field11',
    },
    {
      field_name: 'field12',
    },
  ],
  rows: [
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
    {
      field1: 'test111111111211111111111',
      field2: 'test111111111211111111111',
      field3: 'test111111111211111111111',
      field4: 'test111111111211111111111',
      field5: 'test111111111211111111111',
      field6: 'test111111111211111111111',
      field7: 'test111111111211111111111',
      field8: 'test111111111211111111111',
      field9: 'test111111111211111111111',
      field10: 'test111111111211111111111',
      field11: 'test111111111211111111111',
      field12: 'test111111111211111111111',
    },
  ],
  sql: 'select * from user1',
  start_line: 1,
};

export const queryResShow: SqlQueryResultType[] = [
  {
    hide: false,
    resultItem: query_res_1,
    sqlQueryId: 'query_id_1',
  },
  {
    hide: false,
    resultItem: query_res_1,
    sqlQueryId: 'query_id_2',
  },
];
export const queryResHide_1: SqlQueryResultType[] = [
  {
    hide: true,
    resultItem: query_res_1,
    sqlQueryId: 'query_id_1',
  },
];
export const queryResHide_2: SqlQueryResultType[] = [];
