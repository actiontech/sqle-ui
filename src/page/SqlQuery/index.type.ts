import { IGetSQLResultResDataV1 } from '../../api/common';

export type SqlQueryResultType = {
  sqlQueryId: string;
  resultItem: IGetSQLResultResDataV1;
  hide: boolean;
};

export interface ISqlInputForm {
  instanceName: string;
  instanceSchema: string;
  sql: string;
  maxPreQueryRows: number;
}
