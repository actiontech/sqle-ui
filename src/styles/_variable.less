:root {
  --global-white: #ffffff;
  --global-white-rgb: 255, 255, 255;
  --global-blue-dark: #1e2045;
}

@bg-blue-dark: var(--global-blue-dark);

@bg-white: var(--global-white);
@bg-white-rgb: var(--global-white-rgb);
