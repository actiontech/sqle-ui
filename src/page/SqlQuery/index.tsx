import { WarningOutlined } from '@ant-design/icons';
import { useTheme } from '@mui/styles';
import { useRequest } from 'ahooks';
import { Card, PageHeader, Space, Typography } from 'antd';
import { useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import configuration from '../../api/configuration';
import EmptyBox from '../../components/EmptyBox';
import { OPEN_CLOUD_BEAVER_URL_PARAM_NAME } from '../../data/common';
import { Theme } from '@mui/material/styles';

const SqlQueryEE = () => {
  const { t } = useTranslation();
  const theme = useTheme<Theme>();
  const location = useLocation();
  const cloudbeaverUrl = useRef('');

  const {
    data,
    loading,
    runAsync: getSqlQueryUrl,
  } = useRequest(
    () => {
      return configuration.getSQLQueryConfiguration().then((res) => {
        return res.data.data;
      });
    },
    {
      manual: true,
    }
  );

  useEffect(() => {
    getSqlQueryUrl().then((res) => {
      if (res?.enable_sql_query) {
        cloudbeaverUrl.current = res.sql_query_root_uri as string;
      }

      const params = new URLSearchParams(location.search);

      if (params.get(OPEN_CLOUD_BEAVER_URL_PARAM_NAME) === 'true') {
        window.open(cloudbeaverUrl.current);
      }
    });
  }, [getSqlQueryUrl, location.search]);

  const openCloudbeaver = () => {
    window.open(cloudbeaverUrl.current);
  };

  return (
    <>
      <PageHeader title={t('sqlQuery.pageTitle')} ghost={false}>
        {t('sqlQuery.pageDescribe')}
      </PageHeader>
      <section className="padding-content">
        <Space
          size={theme.common.padding}
          direction="vertical"
          className="full-width-element"
        >
          <Card loading={loading}>
            <EmptyBox
              if={!data?.enable_sql_query}
              defaultNode={
                <Typography.Link onClick={openCloudbeaver}>
                  {t('sqlQuery.jumpToCloudbeaver')}
                </Typography.Link>
              }
            >
              <Space>
                <WarningOutlined
                  className="text-orange"
                  style={{ fontSize: 50 }}
                />
                <div>
                  {t('sqlQuery.eeErrorTips')}ã€‚{t('sqlQuery.eeErrorTips2')}
                  <a
                    target="_blank"
                    href="https://actiontech.github.io/sqle-docs-cn/3.modules/4.2_sql_editor/overview.html"
                    rel="noreferrer"
                  >
                    {t('common.clickHere')}
                  </a>
                </div>
              </Space>
            </EmptyBox>
          </Card>
        </Space>
      </section>
    </>
  );
};

export default SqlQueryEE;
