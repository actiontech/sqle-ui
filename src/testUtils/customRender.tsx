import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider } from '@material-ui/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option]: [...RenderParams]
) => {
  return render(
    <MemoryRouter>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithTheme = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};
