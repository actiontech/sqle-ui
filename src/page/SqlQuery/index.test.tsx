import { ThemeProvider } from '@material-ui/styles';
import SqlQuery from '.';
import { renderWithRedux } from '../../testUtils/customRender';
import lightTheme from '../../theme/light';

describe('SqlQuery', () => {
  test('should match snapshot', () => {
    const { container } = renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlQuery />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
