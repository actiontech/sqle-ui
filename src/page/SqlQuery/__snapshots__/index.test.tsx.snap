// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SqlQuery should match snapshot 1`] = `
<div>
  <div
    class="ant-page-header"
  >
    <div
      class="ant-page-header-heading"
    >
      <div
        class="ant-page-header-heading-left"
      >
        <span
          class="ant-page-header-heading-title"
          title="sqlQuery.pageTitle"
        >
          sqlQuery.pageTitle
        </span>
      </div>
    </div>
    <div
      class="ant-page-header-content"
    >
      sqlQuery.pageDescribe
    </div>
  </div>
  <section
    class="padding-content"
  >
    <div
      class="ant-space ant-space-vertical full-width-element"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 24px;"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-body"
          >
            sqlQuery.ceTips
            <div
              class="ant-typography"
            >
              <ul>
                <li>
                  <a
                    href="https://actiontech.github.io/sqle-docs-cn/"
                  >
                    https://actiontech.github.io/sqle-docs-cn/
                  </a>
                </li>
                <li>
                  <a
                    href="https://www.actionsky.com/"
                  >
                    https://www.actionsky.com/
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 24px;"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                sqlQuery.sqlInput.title
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <form
              class="ant-form ant-form-horizontal"
            >
              <div
                class="ant-row"
                style="margin-left: -12px; margin-right: -12px;"
              >
                <div
                  class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-xl-8 ant-col-xxl-6"
                  style="padding-left: 12px; padding-right: 12px;"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="instanceName"
                        title="sqlQuery.sqlInput.instance"
                      >
                        sqlQuery.sqlInput.instance
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select middle-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                            data-testid="instance-name"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="instanceName_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="instanceName_list"
                                  aria-haspopup="listbox"
                                  aria-owns="instanceName_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="instanceName"
                                  role="combobox"
                                  type="search"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                common.form.placeholder.select
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-xl-8 ant-col-xxl-6"
                  style="padding-left: 12px; padding-right: 12px;"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="instanceSchema"
                        title="sqlQuery.sqlInput.database"
                      >
                        sqlQuery.sqlInput.database
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="instanceSchema_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="instanceSchema_list"
                                  aria-haspopup="listbox"
                                  aria-owns="instanceSchema_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="instanceSchema"
                                  role="combobox"
                                  type="search"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                common.form.placeholder.select
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control makeStyles-editor-3"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        height="200"
                        id="sql"
                        language="sql"
                        theme="vs"
                        value="/* input your sql */"
                        width="100%"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 24px;"
                >
                  <div
                    class="ant-row ant-form-item"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="maxPreQueryRows"
                        title=""
                      >
                        <div
                          class="ant-space ant-space-horizontal ant-space-align-center"
                        >
                          <div
                            class="ant-space-item"
                            style="margin-right: 8px;"
                          >
                            <span
                              aria-label="info-circle"
                              class="anticon anticon-info-circle text-orange"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="info-circle"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                />
                                <path
                                  d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            sqlQuery.sqlInput.returnLength
                          </div>
                        </div>
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-input-number"
                          >
                            <div
                              class="ant-input-number-handler-wrap"
                            >
                              <span
                                aria-disabled="true"
                                aria-label="Increase Value"
                                class="ant-input-number-handler ant-input-number-handler-up ant-input-number-handler-up-disabled"
                                role="button"
                                unselectable="on"
                              >
                                <span
                                  aria-label="up"
                                  class="anticon anticon-up ant-input-number-handler-up-inner"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span
                                aria-disabled="false"
                                aria-label="Decrease Value"
                                class="ant-input-number-handler ant-input-number-handler-down"
                                role="button"
                                unselectable="on"
                              >
                                <span
                                  aria-label="down"
                                  class="anticon anticon-down ant-input-number-handler-down-inner"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-input-number-input-wrap"
                            >
                              <input
                                aria-valuemax="100"
                                aria-valuenow="100"
                                autocomplete="off"
                                class="ant-input-number-input"
                                id="maxPreQueryRows"
                                role="spinbutton"
                                step="1"
                                value="100"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                  style="margin-right: 24px;"
                >
                  <div
                    class="ant-row ant-form-item"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-primary"
                            type="submit"
                          >
                            <span>
                              sqlQuery.sqlInput.searchSqlResult
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    disabled=""
                    type="button"
                  >
                    <span>
                      sqlQuery.sqlInput.sqlHistory
                    </span>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                sqlQuery.executeResult.title
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-empty"
            >
              <div
                class="ant-empty-image"
              >
                <svg
                  class="ant-empty-img-default"
                  height="152"
                  viewBox="0 0 184 152"
                  width="184"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <g
                      transform="translate(24 31.67)"
                    >
                      <ellipse
                        class="ant-empty-img-default-ellipse"
                        cx="67.797"
                        cy="106.89"
                        rx="67.797"
                        ry="12.668"
                      />
                      <path
                        class="ant-empty-img-default-path-1"
                        d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                      />
                      <path
                        class="ant-empty-img-default-path-2"
                        d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                        transform="translate(13.56)"
                      />
                      <path
                        class="ant-empty-img-default-path-3"
                        d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                      />
                      <path
                        class="ant-empty-img-default-path-4"
                        d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                      />
                    </g>
                    <path
                      class="ant-empty-img-default-path-5"
                      d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
                    />
                    <g
                      class="ant-empty-img-default-g"
                      transform="translate(149.65 15.383)"
                    >
                      <ellipse
                        cx="20.654"
                        cy="3.167"
                        rx="2.849"
                        ry="2.815"
                      />
                      <path
                        d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
                      />
                    </g>
                  </g>
                </svg>
              </div>
              <div
                class="ant-empty-description"
              >
                No Data
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
